### PR TITLE
Publication contains datamodel version

### PR DESCRIPTION
**Pull request by axthosarouris**

### CODE REVIEW

File: src/main/java/no/unit/nva/model/Publication.java [hunk 1]
```diff
@@ -21,6 +22,8 @@
 public class Publication
         implements WithIdentifier, WithInternal, WithFile, WithMetadata, WithCopy<Publication.Builder> {
 
+    public static final String MODEL_VERSION= "0.11.8";
```
Comment (torarnet) [inline src/main/java/no/unit/nva/model/Publication.java:25]: why is this public? it has a getter

File: src/test/java/no/unit/nva/model/PublicationTest.java [hunk 2]
```diff
@@ -29,6 +30,7 @@
     public static final String PUBLICATION_FRAME_JSON = "src/main/resources/publicationFrame.json";
 
     protected static final ObjectMapper objectMapper = nva.commons.core.JsonUtils.objectMapper;
+    public static final String DATAMODEL_VERSION = "DATAMODEL_VERSION";
```
Comment (torarnet) [inline src/test/java/no/unit/nva/model/PublicationTest.java:33]: why is this public. Is it used outside the test class?

Comment (torarnet) [inline src/test/java/no/unit/nva/model/PublicationTest.java:33]: Explicit define that it is name for a key or is that not necessary?

**Review by torarnet — DISMISSED** [review]

File: src/main/java/no/unit/nva/model/Publication.java [hunk 3]
```diff
@@ -237,6 +241,16 @@ public void setProjects(List<ResearchProject> projects) {
...
     }
 
+    @JsonProperty("modelVersion")
+    public String getModelVersion(){
+        return MODEL_VERSION;
+    }
+
+    @JsonProperty("modelVersion")
+    public void setModelVersion(){
+        //
```
Comment (brinxmat) [inline src/main/java/no/unit/nva/model/Publication.java:251]: State explicit NO-OP if that's your intention

**Review by brinxmat — CHANGES_REQUESTED** [review]

File: src/test/java/no/unit/nva/model/PublicationTest.java [hunk 4]
```diff
@@ -72,6 +74,14 @@ public void updatingDoiRequestStatusWithoutDoiRequestThrowsIllegalStateException
             is(equalTo(Publication.ERROR_MESSAGE_UPDATEDOIREQUEST_MISSING_DOIREQUEST)));
     }
 
+    @Test
+    public void getModelVersionReturnsModelVersionDefinedByGradle()
```
Comment (svenngjeto) [inline src/test/java/no/unit/nva/model/PublicationTest.java:78]: Tests can be/is package private in Junit5

**Review by janbj — DISMISSED** [review]

File: src/test/java/no/unit/nva/model/PublicationTest.java (hunk 4 above)
Comment (axthosarouris) [reply src/test/java/no/unit/nva/model/PublicationTest.java:77]: This class has some protected methods and package private tests will mess the order of the functions (tests first, helper methods afterwards)

**Review by janbj — DISMISSED** [review]

**Review by brinxmat — APPROVED** [review]

**Pull request merged**